Remove empty JavaFX jars from modulepath

JavaFX Gradle plugin leaves JavaFX empty(non-platform)  jars on modulepath.

With the change made in [JDK-8264998](https://bugs.openjdk.java.net/browse/JDK-8264998), keeping empty jars on the modulepath can result in the following exception:

```
> Task :run FAILED
Error occurred during initialization of boot layer
java.lang.module.FindException: Unable to derive module descriptor for ~/.m2/repository/org/openjfx/javafx-controls/17-ea+8/javafx-controls-17-ea+8.jar
Caused by: java.lang.IllegalArgumentException: javafx.controls.17.ea.8: Invalid module name: '17' is not a Java identifier
```

Some explanation on what's going on:

1. If a jar is on Module path and it doesn't has a Automatic-Module-Name, then Java tries to create a module name from the jar
2. This is done by bisecting the jar name and a number in between text can cause issues: https://docs.oracle.com/javase/9/docs/api/java/lang/module/ModuleFinder.html#automatic-modules
3. Therefore, an empty jar for version "17" may work fine with empty jars containing no module-name. However, empty jars for version `17-ea+8` will fail since a number between alphabets is not supported.

In order to adapt the change made in [JDK-8264998](https://bugs.openjdk.java.net/browse/JDK-8264998), empty jars should be removed from the modulepath.

--- a/src/main/java/org/openjfx/gradle/tasks/ExecTask.java
+++ b/src/main/java/org/openjfx/gradle/tasks/ExecTask.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019, 2020, Gluon
+ * Copyright (c) 2019, 2021, Gluon
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -32,6 +32,9 @@ package org.openjfx.gradle.tasks;
 import org.gradle.api.DefaultTask;
 import org.gradle.api.GradleException;
 import org.gradle.api.Project;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
 import org.gradle.api.plugins.ApplicationPlugin;
 import org.gradle.api.tasks.JavaExec;
 import org.gradle.api.tasks.TaskAction;
@@ -48,6 +51,8 @@ import java.util.List;
 import java.util.TreeSet;
 
 public class ExecTask extends DefaultTask {
+
+    private static final Logger LOGGER = Logging.getLogger(ExecTask.class);
 
     private final Project project;
     private JavaExec execTask;
@@ -74,13 +79,21 @@ public class ExecTask extends DefaultTask {
             var definedJavaFXModuleNames = new TreeSet<>(javaFXOptions.getModules());
             if (!definedJavaFXModuleNames.isEmpty()) {
                 RunModuleOptions moduleOptions = execTask.getExtensions().findByType(RunModuleOptions.class);
+
+                // BEGIN: Remove JavaFX empty jars from classpath
+                final FileCollection classpathWithoutJavaFXJars = execTask.getClasspath().filter(
+                        jar -> Arrays.stream(JavaFXModule.values()).noneMatch(javaFXModule -> jar.getName().contains(javaFXModule.getArtifactName()))
+                );
+                final FileCollection javaFXCPWithoutEmptyJars = execTask.getClasspath().filter(jar -> isJavaFXJar(jar, javaFXOptions.getPlatform()));
+                execTask.setClasspath(classpathWithoutJavaFXJars.plus(javaFXCPWithoutEmptyJars));
+                // END
+
                 if (moduleOptions != null) {
+                    LOGGER.info("Modular JavaFX application found");
                     definedJavaFXModuleNames.forEach(javaFXModule -> moduleOptions.getAddModules().add(javaFXModule));
                 } else {
-                    var javaFXModuleJvmArgs = List.of(
-                            "--module-path", execTask.getClasspath()
-                                    .filter(jar -> isJavaFXJar(jar, javaFXOptions.getPlatform()))
-                                    .getAsPath());
+                    LOGGER.info("Non-modular JavaFX application found");
+                    var javaFXModuleJvmArgs = List.of("--module-path", javaFXCPWithoutEmptyJars.getAsPath());
 
                     var jvmArgs = new ArrayList<String>();
 

